### PR TITLE
Test plan updates & stability levels

### PR DESCRIPTION
**Pull request by luck3y**
F2F: WildFly Community Feature Test Plan

Note this is an initial draft for these changes, wider input is more than welcome!

FYI: @yersan @OndrejKotek @ehsavoie

### CODE REVIEW

File: design-doc-template.adoc [hunk 1]
```diff
@@ -112,6 +112,23 @@ or a confirmation that there are no security implications to consider.
...
+
+** Community - this level should include everything in the 'Preview' stability level, plus the following additional testing as relevant:
+*** Manual tests: briefly describe checks to be performed during one-time exploratory testing. The purpose of this testing is to check corner cases and other cases that are not worth implementing as automated tests. Typical checks are: bad configurations are easy to reveal, attribute descriptions and error messages are clear, names are descriptive and consistent with similar resources, default values are reasonable.
+*** Unit / Integration tests: Manual checks for significant changes in server performance, memory and disk footprint should be described here. These checks are not always relevant, but consideration of these impacts, and others, are strongly encouraged and should be described here. Fully qualified test case names should be provided along with a brief description of what the test is doing.
```
Comment (OndrejKotek) [inline design-doc-template.adoc:126]: "Unit / Integration tests" -> "Miscellaneous checks" ?

**Review by ehsavoie — APPROVED** [review]

**Comment by luck3y** [comment]
FYI: @darranl @bstansberry @jmesnil

**Comment by luck3y** [comment]
Note, converted from draft as there appears to be no more changes and a reminder to all to have a look.

**Review by OndrejKotek — APPROVED** [review]

**Review by yersan — APPROVED** [review]

File: design-doc-template.adoc [hunk 2]
```diff
@@ -112,6 +112,23 @@ or a confirmation that there are no security implications to consider.
 
 == Test Plan
 
+Depending on the selected stability level, the appropriate section below should be completed, including a brief description of how testing is to be performed in accordance with the selected stability level. The non-relevant sections may be removed as needed.
+////
+Depending on the stability level, the test plan required may vary. see below:
+////
+
+** Experimental - no test plan is required. Unit / integration tests may still be added as needed during development.
```
Comment (bstansberry) [inline design-doc-template.adoc:120]: s/may/should

Comment (bstansberry) [reply design-doc-template.adoc:120]: Perhaps

"Basic unit / integration tests should be added during development."

IOW people should write some tests but we're not going to be prescriptive about the details.

**Review by bstansberry — CHANGES_REQUESTED** [review]

**Comment by luck3y** [comment]
@bstansberry I've been out for a couple of days, catching up now, will mail the list and solicit feedback etc.

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @luck3y
